### DISCUSSION TITLE
What's coming up in Feature Release v1.1

### DISCUSSION THREAD

**Discussion by chrislo27** [chrislo27/PolyrhythmMania#31] [Ideas]
Hi folks, I hope you're enjoying Polyrhythm Mania so far. Thanks to community support, I'm excited to share with you what's in store for Feature Release v1.1!

Please note that everything shown below is still subject to change.

* **Polyrhythm: Assemble side mode**
  * The competitor factory is hiring — but it doesn't come easy. Try your hand at assembling widgets in this...particular manner.
  * [Watch the trailer video on YouTube!](https://www.youtube.com/watch?v=zsmNYD7X37Q)
  * ![image](https://user-images.githubusercontent.com/6299069/133912957-b9855fe8-ef9c-43ff-a2d3-cad34cc5ec97.png)
* **Daily Challenge online leaderboard**
  * See the community's high scores in the Daily Challenge!
  * ![image](https://user-images.githubusercontent.com/6299069/133912962-61f91689-9a54-4842-a544-1b277426fd02.png)
* **Level Metadata**
  * Levels will get info fields to aid with sorting and organizing in Library view (below)
  * ![image](https://user-images.githubusercontent.com/6299069/133912968-37d62d57-b93e-49af-b6e6-3b623209b2f8.png)
* **Library view**
  * Select folders for the game to search, and sort and organize your levels based on its Level Metadata!
* **Separating Project and Level files**
  * Project files and Level files will be separated
  * Project files have the new file extension `.prmproj`
    * They represent an unfinished level that may not exactly be ready to play
  * Level files will keep the file extension `.prmania`
    * They can be "published" from project files, and will mandate certain Level Metadata
    * They may also be checked to make sure they are playable and maybe to generate level stats
    * Older `.prmania` level files will still be playable
* **Text box improvements**
  * You will be able to change the Style of your text box
  * ![image](https://user-images.githubusercontent.com/6299069/133913004-6433be97-de65-411f-a446-74505bb01a86.png) 
![image](https://user-images.githubusercontent.com/6299069/133913010-4da1b2aa-de47-48de-b46c-791657598218.png)
* **Song Info Cards**
  * These helpful on-screen prompts make a return from RHRE and Bouncy Road Mania
  * They can provide text directly from Level Metadata, or you can provide your own text
  * ![image](https://user-images.githubusercontent.com/6299069/133913038-897b1877-5fa5-46c0-b8dd-efd2e898c25a.png)

Let me know what you think of these planned features so far! And of course, a huge thank you to the community for the immense amount of support so far — it's far beyond what I ever imagined out of this side project. If you'd like to support the project's development, [consider a small donation](https://www.paypal.com/donate?hosted_button_id=9JLGHKZNWLLQ8) or even just Starring the repository (top right corner).

**Comment by TerrariaTree** [comment]
I love this!!!! I can't wait for the Assemble mode, and leaderboard looks exciting!

**Comment by GierkowiecTV** [comment]
Such a good of an update by look of things!
but i also have a few other ideas that maybe will be added in 1.2v if that one will even exist XD:
edit mode:
a ruler, save your remix as an mp3!, an option to change the textures for Try Again, OK, and SuperB, a possibility to change the value for 
when ok try again and superb appear, swing, 
play a prmania level:
karate man 2 mode (randomly switches the speed every 2 time signatures (idk how to explain it, if it's 4/4 then every 8 beats or 3/4 then every 6 beats)
blindless mode 
disable custom values for result screen (if there are any)
other:
editor for polyrhythm dunk polyrhythm assamble and other bonus games
make your own endless mode! (create max 15 patters that would appear during your own endless mode)
vsync mode
a tutorial for the bonus modes (it can even be just a few sentences about what you need to do)
that's all i have i hope you'll like some of those ideas, i know they aren't that hard to implement into the game (besides editor for polyrhythm dunk and polyrhythm assemble) and have a good day

**Comment by chrislo27** [comment]
I'd like to share some progress so far. Note that everything shown is still a work in progress and will change for the full release.

Level Metadata is about the same as was previously shown:
![image](https://user-images.githubusercontent.com/6299069/138044073-88ff313f-d1b3-4de6-97e2-65a0cf20ebf6.png)

You can now add a Banner image to your level:
![image](https://user-images.githubusercontent.com/6299069/138044105-dc3dddaa-f88c-41f4-bd3d-13cd2164d107.png)

...But remember to export your level first so it can be checked for errors and have other statistics saved:
![image](https://user-images.githubusercontent.com/6299069/138044177-4c1ea64a-cfd7-4478-891c-c23339062105.png)

This is an extremely early version of the Library menu. Isn't it nice to have details of your level collection easily accessible? Legacy levels don't have metadata information, but you can easily update them (better yet, ask the original level creator to update their level).
![image](https://user-images.githubusercontent.com/6299069/138044308-ef183ca5-5837-4c74-8d3f-814fdf154d02.png)


[You can watch the Azalea Town level I'm using as a test, made by Turtike, here on YouTube.](https://www.youtube.com/watch?v=f-nJhTg6pFc) It is an *extremely good* level, making full use of custom texture packs!

There's still *lots* of work to be done, but I think the Level Metadata and Library features will be a significant upgrade to the user experience and will be well worth it. I haven't been able to work on this as much as I'd like, so I appreciate your patience. I also have some other things planned for v1.1 but I'll keep those a surprise for now. :)
